src/briefcase/integrations/subprocess.py

```diff
@@ -197,10 +204,51 @@ def run(self, args, **kwargs):
         except subprocess.CalledProcessError as e:
             self._log_return_code(e.returncode)
             raise
-
         self._log_return_code(command_result.returncode)
+
         return command_result
 
+    def _run_in_wait_bar(self, args, check=False, **kwargs):
```

Comment:
This implementation is (as much as possible) a mirror of `subprocess.run`.....albeit a bit simplified. A lot of the complexity is driven by `timeout` anyway.

Reply:
I think I follow what's going on here; but just to make sure my understanding is correct:

When we're in a wait bar, there's an obvious conflict between stdout and displaying the wait bar. This conflict is avoided as long as we *only* use Log/Console for output. To make sure that happens, we intercept any use of `run()` while we're in a waitbar context, and insert an output streamer, instead of letting `run()` output directly to stdout. This ensures that the stdout/stderr content is still visible, but also guarantees the integrity of the waitbar rendering.

Have I got this correct?

Reply:
That is the correct understanding.

> This conflict is avoided as long as we only use Log/Console for output.

Yes; although, more precisely, we need to make sure the same instance of `rich.console.Console` is being used for any printing to `stdout`. Given that the instantiated Rich Console in `console.py` is effectively a singleton, even multiple instances of `Log` or `Console` should work properly together (although, I don't think multiple instances should manifest in normal briefcase operation). I bring it up, though, since I was originally having problems with screen corruption because I was instantiating some Rich classes without explicitly passing the Rich Console that briefcase was using. This may be important to keep in mind if additional Rich concepts are incorporated in the future.